composer/trainer/trainer.py

```diff
@@ -308,7 +310,10 @@ def __init__(
             if self.deepspeed_enabled:
                 raise NotImplementedError("Checkpointing is not yet supported with DeepSpeed.")
             self.checkpoint_loader = CheckpointLoader(checkpoint_filepath=checkpoint_filepath)
-            self.checkpoint_loader.load_checkpoint(state=self.state)
+            restored_seed = self.checkpoint_loader.load_checkpoint(state=self.state)
+            if restored_seed is not None:
```

Comment:
This was necessary to handle the case:
- Train a model and save a checkpoint which includes the seed
- Resume from checkpoint and use the saved seed to seed all rng objects
- Save a checkpoint again (in this case `self.seed` is stored in the checkpoint so it must be the correct (i.e. originally checkpointed) seed
- Resume from checkpoint

Reply:
This is really helpful...I'd add this as a comment in the code